sdk/storage/src/core/mod.rs

```diff
@@ -42,77 +43,83 @@ use serde::{Deserialize, Deserializer};
 pub use stored_access_policy::{StoredAccessPolicy, StoredAccessPolicyList};
 
 #[derive(Debug, Clone, PartialEq)]
-pub struct ConsistencyCRC64(pub [u8; 8]);
+pub struct ConsistencyCRC64(pub Bytes);
+
+impl ConsistencyCRC64 {
+    /// Decodes from base64 encoded input
+    pub fn decode(input: impl AsRef<[u8]>) -> Result<Self, Error> {
+        let bytes = base64::decode(input).map_err(Error::Base64DecodeError)?;
+        let bytes = Bytes::from(bytes);
+        match bytes.len() {
+            8 => Ok(Self(bytes)),
+            len => Err(Error::CRC64Not8BytesLong(len)),
+        }
+    }
+}
 
 impl<'de> Deserialize<'de> for ConsistencyCRC64 {
     fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
     where
         D: Deserializer<'de>,
     {
-        String::deserialize(deserializer)
-            .and_then(|str| base64::decode(str).map_err(serde::de::Error::custom))
-            .and_then(|content_crc64_vec| {
-                if content_crc64_vec.len() != 8 {
-                    return Err(serde::de::Error::custom(crate::Error::CRC64Not8BytesLong(
-                        content_crc64_vec.len() as u64,
-                    )));
-                }
-                let mut content_crc64 = [0; 8];
-                content_crc64.copy_from_slice(&content_crc64_vec[0..8]);
-                Ok(Self(content_crc64))
-            })
+        let bytes = <&[u8]>::deserialize(deserializer)?;
```

Comment:
This breaks XML deserialization.

Reply:
It does. Thanks for the new tests! I changed it back to StringDeserializer like you had.